### ISSUE TITLE
Upstream ResponseWriter of type *headers.responseModifier does not implement http.CloseNotifier

### ISSUE THREAD

**Issue by jtremesay-sereema** [traefik/traefik#8222]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Traefik used as reverse proxy for a Django application, the stack running on docker swarm. Don’t know when the problem started, we only keep one week of logs, so "at least a week".

### What did you expect to see?

No warnings

### What did you see instead?

A lot of warnings about an object ResponseWriter no implementing an interface (I think, I don’t know golang much)

```
2021-06-24T07:15:02.843669+00:00 sereema_traefik: time="2021-06-24T07:15:02Z" level=warning msg="Upstream ResponseWriter of type *headers.responseModifier does not implement http.CloseNotifier. Returning dummy channel."
```

I have found similar error message on others issues like traefik/traefik#2628, but we don’t have internal server error or stuff like that, just the warnings.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.4.9
Codename:     livarot
Go version:   go1.16.5
Built:        2021-06-21T16:17:58Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

docker-compose.yaml:

```YAML
version: '3.8'

networks:
    redis:
        driver: overlay
    traefik:
        driver: overlay

services:
    traefik:
        user: ${CONTAINER_UID:-0}
        build: traefik
        image: ${TRAEFIK_IMAGE:-traefik}
        deploy:
            placement:
                constraints:
                    - "node.role==manager"
            #resources:
            #    limits:
            #        cpus: '0.5'
            #        memory: 200M
        healthcheck:
            test: ["CMD", "traefik", "healthcheck"]
            interval: 30s
            timeout: 5s
        environment:
            - AWS_REGION
            - AWS_HOSTED_ZONE_ID
            - MAIN_HOST
            - LEGACY_HOSTS
            - TRAEFIK_LOG_LEVEL=INFO
            - TRAEFIK_METRICS_STATSD=true
            # This is *for now* the host op for the cloudwatch statsd server
            - TRAEFIK_METRICS_STATSD_ADDRESS=172.17.0.1:8125
            - TRAEFIK_ENTRYPOINTS_WEBUNSAFE=true
            - TRAEFIK_ENTRYPOINTS_WEBUNSAFE_ADDRESS=:1080
            - TRAEFIK_ENTRYPOINTS_WEB=true
            - TRAEFIK_ENTRYPOINTS_WEB_ADDRESS=:10443
            - TRAEFIK_SERVERSTRANSPORT_FORWARDINGTIMEOUTS_DIALTIMEOUT=10
            - TRAEFIK_SERVERSTRANSPORT_FORWARDINGTIMEOUTS_IDLECONNTIMEOUT=90
            - TRAEFIK_SERVERSTRANSPORT_FORWARDINGTIMEOUTS_RESPONSEHEADERTIMEOUT=90
            - TRAEFIK_SERVERSTRANSPORT_MAXIDLECONNSPERHOST=4
            - TRAEFIK_PROVIDERS_FILE_FILENAME=/etc/traefik/traefik_dynamic.toml
            - TRAEFIK_CERTIFICATESRESOLVERS_LE=true
            - TRAEFIK_CERTIFICATESRESOLVERS_LE_ACME_EMAIL=kevin.michel@sereema.com
            - TRAEFIK_CERTIFICATESRESOLVERS_LE_ACME_STORAGE=/acme/acme-ec256.json
            - TRAEFIK_CERTIFICATESRESOLVERS_LE_ACME_CASERVER=${TRAEFIK_CASERVER:-https://acme-staging-v02.api.letsencrypt.org/directory}
            - TRAEFIK_CERTIFICATESRESOLVERS_LE_ACME_DNSCHALLENGE=true
            - TRAEFIK_CERTIFICATESRESOLVERS_LE_ACME_DNSCHALLENGE_PROVIDER=route53
            # Traefik is super-slow to validate all domains if it doesn't have certificates in cache.
            # If you juste created a new server and get weird TLS errors, it's probably traefik
            # replying in basic http over https port because certificates aren't ready yet...
            - TRAEFIK_CERTIFICATESRESOLVERS_LE_ACME_DNSCHALLENGE_RESOLVERS=1.1.1.1:53,8.8.8.8:53
            # I'd rather use NSA-free RSA keys instead of EC256 but IE11 on
            # Win7 and Win8.1 does not support GCM suites with RSA keys.
            # We could keep RSA keys and enable CBC cipher suites but CBC
            # attacks are a known fact (even with TLS>=1.2) whereas EC256
            # NSA-injected weakness is only theoretical.
            # According to netmarketshare.com, as of 2020-04, IE11 still has
            # around 6% of desktop market share and 30% of it are still Win7
            # and Win8.1 (the desktop market share is now 40% of the whole
            # market): in total we could lose around 0.7% of our visitors by
            # ignoring this issue. We have UserAgent stats in the product to
            # measure our own audience but we must first allow IE11 to gather
            # unbiased visitor stats.
            - TRAEFIK_CERTIFICATESRESOLVERS_LE_ACME_KEYTYPE=EC256
            - TRAEFIK_PING=true
            - TRAEFIK_PING_ENTRYPOINT=webunsafe
            - WEB_BACKEND_URL=http://web:8000/
        networks:
            - traefik
        ports:
            -   target: 1080
                published: 80
                protocol: tcp
                mode: host
            -   target: 10443
                published: 443
                protocol: tcp
                mode: host
        volumes:
            -   type: bind
                # We're still relying on the host, again with the userid problem...
                source: ${ACME_VOLUME_SOURCE:-/var/cache/acme}
                target: /acme
    web:
        user: ${CONTAINER_UID:-0}
        build: ..
        image: ${WINDFIT_IMAGE:-windfit}
        deploy:
            placement:
                constraints:
                    - "node.role==worker"
            replicas: 2
            #resources:
            #    limits:
            #        cpus: '0.5'
            #        memory: 600M
            update_config:
                delay: 10s
                order: start-first
        healthcheck:
            test: [ "CMD", "python3", "-c", "import urllib.request;print(urllib.request.urlopen('http://127.0.0.1:8000/ping').getcode())" ]
            interval: 30s
            timeout: 5s
        command: gunicorn
            --bind 0.0.0.0:8000
            --keep-alive 60
            --preload
            --workers 2
            --worker-class uvicorn.workers.UvicornWorker
            sereema.asgi
        environment:
            - ASGI_THREADS=4
            - AWS_SERVER_USERNAME
            - C2_SERVICE_HOST
            - DATABASE_URL
            - DATABASE_REPLICA_URLS
            - MAIN_HOST
            - LEGACY_HOSTS
            - DJANGO_SECRET_KEY
            - REDIS_PASSWORD
            - CHANNELS_SECRET_KEY
            - SENDGRID_API_KEY
            - SENTRY_DSN
            - WINDFIT_STATSD_ADDRESS=172.17.0.1
        networks:
            - traefik
            - redis
```

dynamic.toml:

```toml
[http]
  [http.routers]
    [http.routers.web]
      entryPoints = ["web"]
      rule = "Host(`{{ env "MAIN_HOST" }}`{{ range env "LEGACY_HOSTS" | splitList " " | compact }},`{{ . }}`{{ end }})"
      service = "web"
      middlewares = ["noserver", "inflight", "limit", "retry"]

      [http.routers.web.tls]
        certResolver = "le"
        options = "default"

        [[http.routers.web.tls.domains]]
          main = "{{ env "MAIN_HOST" }}"
          sans = [
            {{ range env "LEGACY_HOSTS" | splitList " " | compact }}"{{ . }}", {{ end }}
          ]

    [http.routers.webunsafe]
      entryPoints = ["webunsafe"]
      rule = "Host(`{{ env "MAIN_HOST" }}`{{ range env "LEGACY_HOSTS" | splitList " " | compact }},`{{ . }}`{{ end }})"
      service = "web"
      middlewares = ["noserver", "inflight", "limit", "retry"]

    [http.routers.links]
      entryPoints = ["web"]
      rule = "Host(`links.sereema.com`)"
      service = "links"

      [http.routers.links.tls]
        certResolver = "le"
        options = "default"

        [[http.routers.links.tls.domains]]
          main = "links.sereema.com"

    [http.routers.linksunsafe]
      entryPoints = ["webunsafe"]
      rule = "Host(`links.sereema.com`)"
      service = "links"

  [http.services]
    [http.services.web]
      [http.services.web.loadBalancer]
        [[http.services.web.loadBalancer.servers]]
          url = "{{ env "WEB_BACKEND_URL" }}"

    [http.services.links]
      [http.services.links.loadBalancer]
        [[http.services.links.loadBalancer.servers]]
          url = "https://sendgrid.net"

  [http.middlewares]
    [http.middlewares.noserver]
      [http.middlewares.noserver.headers]
        [http.middlewares.noserver.headers.customResponseHeaders]
          server = ""

    [http.middlewares.retry]
      [http.middlewares.retry.retry]
        attempts = 4

    [http.middlewares.limit]
      [http.middlewares.limit.buffering]
        maxRequestBodyBytes = 20000000 # 20Mb
        memRequestBodyBytes = 2000000 # 2Mb

    [http.middlewares.inflight]
      [http.middlewares.inflight.inFlightReq]
        # This is below half our soft ulimit on open files
        amount = 500

[tls]
  [tls.options]
    [tls.options.default]
      sniStrict = true
      minVersion = "VersionTLS12"
      cipherSuites = [
        "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
        "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
        "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
        "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
        "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305",
        "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305"
      ]
```

We use django middlewares like the [GZipMiddleware](https://docs.djangoproject.com/en/3.2/ref/middleware/#module-django.middleware.gzip), maybe something related to HTTP headers modification?

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

I cannot reproduce locally, and if not needed, I prefer not enabling debug log level on the prod.

**Comment by dtomcej** [comment]
Hello @jtremesay-sereema,

That error can be caused when a client closes a connection before the server has responded, and the underlying transport is closed.

Have you noticed any unwanted behaviour outside of the logging?

**Comment by jtremesay-sereema** [comment]
We don’t notice any unwanted behaviour outside of the logging.

**Comment by traefiker** [comment]
Closed by #8238.
